### PR TITLE
Update Predator Bot

### PR DESCRIPTION
**Pull request by Sparks1992**
Added Stone Hide skill usage after using Venomous Leap skill
Added Stone Hide skill usage when retreating

### CODE REVIEW

File: bots/heroes/predator_main.lua [hunk 1]
```diff
@@ -276,6 +277,8 @@ local function HarassHeroExecuteOverride(botBrain)
 			if bShouldLeap then
 				if bDebugEchos then BotEcho('LEAPIN!') end
 				bActionTaken = core.OrderAbilityEntity(botBrain, leap, unitTarget)
+					if stonehide:CanActivate() then
+						bActionTaken = core.OrderAbility(botBrain, stonehide)
```
Comment (Anakonda) [inline bots/heroes/predator_main.lua]: This if is missing end. That is syntax error - do you even test these?

File: bots/heroes/predator_main.lua [hunk 2]
```diff
@@ -288,7 +291,14 @@ end
...
 behaviorLib.HarassHeroBehavior["Execute"] = HarassHeroExecuteOverride
 
-
+-----------------------
+-- Return to well
+-----------------------
+--this is a great function to override with using retreating skills, such as blinks, travels, stuns or slows.
+function behaviorLib.CustomReturnToWellExecute(botBrain)
+	local stonehide = skills.abilStoneHide
+	return  core.OrderAbility(botBrain, stonehide)
```
Comment (Anakonda) [inline bots/heroes/predator_main.lua]: Check if it can be activated, also only cast it if there are enemies

**Review by Anakonda — CHANGES_REQUESTED** [review]

**Comment by Sparks1992** [comment]
Changed both issues:
- added the end (sorry, can't test them really well since there is no test+++ button in practice with bots)
- added bCanSee to retreat use of StoneHide

**Comment by Anakonda** [comment]
open console with ctrl+f8 run command ´showwidget testppp_mainButton´ and close console with ctrl+f8. This should give you the test+++ button.
unitTarget is the hero you last tried to kill it can be anywhere in the map by the time you are retreating - like visible on another lane. You need to use the localunits table, see Moon Queens ult if you need exampl.e

**Comment by Sparks1992** [comment]
I updated it, i tried to add Predator in bots folder but i think it's overwritten by the one from the game, because in the bots pick it appears only one Predator, and that's the one from the game.

**Comment by Sparks1992** [comment]
done, it works...but it activates too late...first he retreats without using the skill and he starts using it only below 15-20% after i chase him for a couple of seconds. How could i make him use it as soon as he is starting to retreat?

File: bots/heroes/predator_main.lua [hunk 3]
```diff
@@ -288,7 +292,47 @@ end
...
+	for id, unitEnemy in pairs(tEnemies) do
+		if core.CanSeeUnit(botBrain, unitEnemy) then
+			nCount = nCount + 1
+		end
+	end
+	
+	if unitSelf:GetHealthPercent() < .50 then
+		local stonehide = skills.abilStoneHide
+		-- BACK OFF!
+		if bCanSeeUnit and stonehide:CanActivate() then
```
Comment (Anakonda) [inline bots/heroes/predator_main.lua]: instead of bCanSeeunit check that nCount > 0
Then remove all unitTarget stuff

**Comment by Sparks1992** [comment]
Updated - removed stuff that wasn't used and added nCount > 0 instead of bCanSee

**Review by Anakonda — APPROVED** [review]
Approved

**Pull request merged**